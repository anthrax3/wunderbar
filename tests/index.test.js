const stripAnsi = require('strip-ansi');

const data = require('./__fixtures__/age-distrib-germany.json');
const draw = require('../lib');

describe('draw', () => {
  const outputResult = (data, options) => {
    const { chart, scale, legend } = draw(data, options);
    return '\n' + stripAnsi([chart, scale, legend].join('\n\n')).trim() + '\n';
  };

  it('should provide result with chart, scale and legend', () => {
    expect(outputResult(data, { length: 100 })).toMatchSnapshot();
  });
});
