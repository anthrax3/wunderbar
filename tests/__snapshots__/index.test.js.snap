// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`draw should provide result with chart, scale and legend 1`] = `
"
█████████

████████████████████████████████████████████████████████████████████████████████████████████████████
███████████████
███████████████████████████████████████

◺ 8131788.00                                                                           32601939.00 ◿

█ -  0-14 10333117.00
█ - 15-24  8131788.00
█ - 25-54 32601939.00
█ - 55-64 11747501.00
█ -   65+ 17779672.00
"
`;
